ToolChain/Assembler/Parser/Parser.py

```diff
@@ -136,18 +136,15 @@ def _findInstructionCode(self, form, line):
         except KeyError as e:
             raise ValueError("Invalid instruction format")
 
-        found = False
-
         for possibleCodes in ins:
             # We shift the binary code by 4 bits to see if we have a match with typeCode for this form
             # ex: typeCode = 0010, possiblecodes = 00101011 would be true since the first 4 bits match.
             # Once we have our instruction match, we know the size of the instruction and we add to the counter
             if insform["typeCode"] is (possibleCodes >> 4):
-                found = True
                 self.relativeAddress += insform["length"]
                 instruction += bytes((possibleCodes,))
-
-        if not found:
+                break
```

Comment:
I think you can do : 
```python3
for e in elems:
   if cond: 
       # set instruction
       return instruction
else:
   raise # Err
```
I could be wrong. try it if you think its better and let me know.